week4/trie/thalesmg/Main.hs

```diff
@@ -0,0 +1,56 @@
+import qualified Data.Map.Lazy as M
+
+data Trie a = Empty | Node (M.Map a (Trie a)) deriving (Show)
```

Comment:
Usually the imports go like

```
import qualified Data.Map as M
import Data.Map (Map)
```

and then you don't need `M.` in `M.Map`. It's not something bad, just something I wanted to mention.